controllers/agents/app/serviceclaim_controller.go

```diff
@@ -43,12 +49,124 @@ type ServiceClaimReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (r *ServiceClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	l := log.FromContext(ctx).WithValues("namespace", req.Namespace, "name", req.Name)
+	l.Info("Reconciling service claim")
+
+	var sclaim primazaiov1alpha1.ServiceClaim
+	if err := r.Get(ctx, req.NamespacedName, &sclaim); err != nil {
+		l.Error(err, "Failed to retrieve ServiceClaim")
+		if apierrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	config, remote_namespace, err := workercluster.GetPrimazaKubeconfig(ctx, sclaim.Namespace, r.Client, constants.ApplicationAgentKubeconfigSecretName)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	l.Info("remote cluster", "address", config.Host)
+
+	remote_client, err := client.New(config, client.Options{
+		Scheme: r.Client.Scheme(),
+		Mapper: r.Mapper,
+	})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	objKey := client.ObjectKey{
+		Name:      constants.ApplicationAgentDeploymentName,
+		Namespace: req.NamespacedName.Namespace,
+	}
+	var deployment appsv1.Deployment
+	err = r.Get(ctx, objKey, &deployment)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// it should never happen that this controller does not find itself
+			l.Error(err,
```

Comment:
should be `"application agent deployment not found, that should be a bug",
				"expected deployment name", constants.ApplicationAgentDeploymentName`

Reply:
Fixed.